packages/reports/src/Containers/open-positions-table.tsx

```diff
@@ -0,0 +1,154 @@
+import EmptyTradeHistoryMessage from '../Components/empty-trade-history-message';
+import PlaceholderComponent from '../Components/placeholder-component';
+import { TUnsupportedContractType } from 'Types';
+import { TMobileRowRenderer } from './mobile-row-renderer';
+import { TTotals } from './open-positions';
+import { TPortfolioPosition } from '@deriv/stores/types';
+
+type TEmptyPlaceholderWrapper = React.PropsWithChildren<{
+    is_empty: boolean;
+    component_icon: string;
+}>;
+
+type TOpenPositionsTable = {
+    accumulator_rate: string;
+    active_positions: TPortfolioPosition[];
+    className: string;
+    columns: Record<string, unknown>[];
+    component_icon: string;
+    contract_type_value: string;
+    currency: string;
+    is_empty: boolean;
+    is_loading: boolean;
+    mobileRowRenderer: React.ComponentProps<typeof DataList>['rowRenderer'];
+    row_size: number;
+    totals: TTotals;
+};
+
+const EmptyPlaceholderWrapper = ({ is_empty, component_icon, children }: TEmptyPlaceholderWrapper) => (
+    <React.Fragment>
+        {is_empty ? (
+            <PlaceholderComponent
+                is_empty={is_empty}
+                empty_message_component={EmptyTradeHistoryMessage}
+                component_icon={component_icon}
+                localized_message={<Localize i18n_default_text='You have no open positions yet.' />}
+            />
+        ) : (
+            children
+        )}
+    </React.Fragment>
+);
+
+/*
+ * purchase property in contract positions object can be NaN or undefined in the first few responses.
+ * So we set it to true in these case to show a preloader for the data-table-row until the correct value is set.
+ */
+export const isPurchaseMissing = (item: { purchase?: number }) => isNaN(Number(item.purchase)) || !item.purchase;
+
+export const getRowAction = (row_obj: NonNullable<TMobileRowRenderer['row']> = {}) => {
+    let action: string | { component?: React.ReactElement } = {};
+    const unsupportedContractConfig = getUnsupportedContracts()[row_obj.type as TUnsupportedContractType];
+
+    if (row_obj.contract_info && !hasContractStarted(row_obj.contract_info)) {
+        action = {
+            component: <Localize i18n_default_text="You'll see these details once the contract starts." />,
+        };
+    } else if (unsupportedContractConfig || row_obj.id) {
+        action = unsupportedContractConfig
+            ? {
+                  component: (
+                      <Localize
+                          i18n_default_text="The {{trade_type_name}} contract details aren't currently available. We're working on making them available soon."
+                          values={{
+                              trade_type_name: unsupportedContractConfig?.name,
+                          }}
+                      />
+                  ),
+              }
+            : getContractPath(row_obj.id);
+    }
+
+    return action;
+};
```

Comment:
Note: conditions in this `getRowAction()` helper are refactored. previously, it was located in `open-positions.tsx`